tooling/lib/build.js

```diff
@@ -6,32 +6,104 @@ const {
   transformFile
 } = require(`../lib/async`);
 const path = require(`path`);
-const {writeFile} = require(`fs-promise`);
+const {readFile, writeFile} = require(`fs-promise`);
 const {glob} = require(`../util/package`);
+const {exists} = require(`./fs`);
+const {cloneDeep} = require(`lodash`);
 
-exports.buildFile = async function buildFile({src, dest}) {
+exports.buildFile = buildFile;
+exports.buildPackage = buildPackage;
+
+/**
+ * Build a single file
+ * @param {string} src
+ * @param {string} dest
+ * @param {Object} config
+ * @returns {Promise}
+ */
+async function buildFile(src, dest, config = {}) {
   debug(`transforming ${src}`);
-  const {code, map} = await transformFile(src);
-  debug(`transformFileed ${src}`);
+  const {code, map} = await transformFile(src, config);
+  debug(`transformed file ${src}`);
   await mkdirp(path.dirname(dest));
   debug(`writing ${dest}`);
   await writeFile(dest, code);
   await writeFile(`${dest}.map`, JSON.stringify(map));
   debug(`wrote ${dest}`);
-};
+}
 
-exports.buildPackage = async function buildPackage(packageName) {
+/**
+ * Build all the files in a given package, for node, browser, and
+ * es6-modules(node)
+ * @param {string} packageName
+ * @returns {Promise}
+ */
+async function buildPackage(packageName) {
   debug(`building package ${packageName}`);
   const files = await glob(`src/**/*.js`, {packageName});
   debug(`building files `, files);
-  const mapped = files
-  .map((filename) => path.join(`packages`, `node_modules`, packageName, filename))
-  .map((filename) => ({
-    src: filename,
-    dest: filename.replace(`src`, `dist`)
-  }));
-
-  for (const file of mapped) {
-    await exports.buildFile(file);
+
+  const mapped = [];
+  for (const filename of files) {
+    if (!filename.endsWith(`.browser.js`)) {
+      mapped.push({
+        src: await srcFromFilename(packageName, filename),
+        dest: {
+          browser: path.resolve(`packages`, `node_modules`, packageName, filename.replace(`src`, `dist/browser`)),
+          main: path.resolve(`packages`, `node_modules`, packageName, filename.replace(`src`, `dist/main`)),
+          module: path.resolve(`packages`, `node_modules`, packageName, filename.replace(`src`, `dist/module`))
+        }
+      });
+    }
+  }
+
+  const config = JSON.parse(await readFile(path.join(process.cwd(), `.babelrc`)));
```

Comment:
Would using the sync version of `readFile` (and other fs methods) make more sense than doing async/await?